pkg/distributors/contracts/MultiRewards.sol

```diff
@@ -389,16 +394,23 @@ contract MultiRewards is IMultiRewards, IDistributor, ReentrancyGuard, Temporari
     /* ========== RESTRICTED FUNCTIONS ========== */
 
     /**
-     * @notice Allows a rewards distributor to deposit more tokens to be distributed as rewards
+     * @notice Allows a rewards distributor, or the reward scheduler
+     * to deposit more tokens to be distributed as rewards
+     * @param pool - the pool bpt that is staked in this contract
      * @param rewardsToken - the token to deposit into staking contract for distribution
      * @param reward - the amount of tokens to deposit
      */
     function notifyRewardAmount(
         IERC20 pool,
         IERC20 rewardsToken,
-        uint256 reward
+        uint256 reward,
+        address rewarder
     ) external override updateReward(pool, address(0)) {
-        require(_rewarders[pool][rewardsToken].contains(msg.sender), "Reward must be configured with addReward");
+        require(
+            msg.sender == rewarder || msg.sender == address(rewardsScheduler),
+            "Rewarder must be sender, or rewards scheduler"
+        );
+        require(_rewarders[pool][rewardsToken].contains(rewarder), "Reward must be configured with addReward");
 
         // handle the transfer of reward tokens via `safeTransferFrom` to reduce the number
         // of transactions required and ensure correctness of the reward amount
```

Comment:
I'd add a comment here mentioning than the reward scheduler holds the tokens for `rewarder`, so we always use `msg.sender` as the sender address.